src/common/QskRgbValue.h

```diff
@@ -236,4 +236,312 @@ namespace QskRgb
+                    break;
+
+                case '5':
+                    rgb = ( rgb << 4 ) | 0x5;
+                    break;
+
+                case '6':
+                    rgb = ( rgb << 4 ) | 0x6;
+                    break;
+
+                case '7':
+                    rgb = ( rgb << 4 ) | 0x7;
+                    break;
+
+                case '8':
+                    rgb = ( rgb << 4 ) | 0x8;
+                    break;
+
+                case '9':
+                    rgb = ( rgb << 4 ) | 0x9;
+                    break;
+
+                case 'A':
+                case 'a':
+                    rgb = ( rgb << 4 ) | 0xA;
+                    break;
+
+                case 'B':
+                case 'b':
+                    rgb = ( rgb << 4 ) | 0xB;
+                    break;
+
+                case 'C':
+                case 'c':
+                    rgb = ( rgb << 4 ) | 0xC;
+                    break;
+
+                case 'D':
+                case 'd':
+                    rgb = ( rgb << 4 ) | 0xD;
+                    break;
+
+                case 'E':
+                case 'e':
+                    rgb = ( rgb << 4 ) | 0xE;
+                    break;
+
+                case 'F':
+                case 'f':
+                    rgb = ( rgb << 4 ) | 0xF;
+                    break;
+
+                default:
+                    rgb = ( rgb << 4 ) | 0x0;
+                    break;
+            }
+        }
+
+        return rgb;
+    }
+
+    namespace literals
+    {
+        template< QRgb V >
+        constexpr QRgb parse_hex_literal_h()
+        {
+            return V;
+        }
+
+        template< QRgb V, char C, char... Cs >
+        constexpr QRgb parse_hex_literal_h()
+        {
+            constexpr auto is_0_9 = C >= '0' && C <= '9';
+            constexpr auto is_a_f = C >= 'a' && C <= 'f';
+            constexpr auto is_A_F = C >= 'A' && C <= 'F';
+
+            static_assert( is_0_9 || is_a_f || is_A_F, "invalid hex character" );
+
+            if constexpr ( is_0_9 )
+            {
+                return parse_hex_literal_h< ( V << 4 ) | ( C - '0' ), Cs... >();
+            }
+            if constexpr ( is_a_f )
+            {
+                return parse_hex_literal_h< ( V << 4 ) | ( C - 'a' + 10 ), Cs... >();
+            }
+            if constexpr ( is_A_F )
+            {
+                return parse_hex_literal_h< ( V << 4 ) | ( C - 'A' + 10 ), Cs... >();
+            }
+        }
+
+        template< char C, char... Cs >
+        constexpr QRgb parse_hex_literal_x()
+        {
+            static_assert( C == 'x', "invalid hex prefix character" );
+            return parse_hex_literal_h< 0, Cs... >();
+        }
+
+        template< char C, char... Cs >
+        constexpr QRgb parse_hex_literal_0()
+        {
+            static_assert( C == '0', "invalid hex prefix character" );
+            return parse_hex_literal_x< Cs... >();
+        }
+
```

Comment:
Why not ( there is probably a better name than "fromHexRgba" ):

```
        constexpr QRgb fromHexRgba(
            const char* const str, const size_t len ) noexcept
        {
            constexpr auto max = 9;
            const auto argb = static_cast< int >( fromHexString( str, len ) );
            const auto r = ( argb >> ( len == max ? 24 : 16 ) ) & 0xFF;
            const auto g = ( argb >> ( len == max ? 16 : 8 ) ) & 0xFF;
            const auto b = ( argb >> ( len == max ? 8 : 0 ) ) & 0xFF;
            const auto a = ( len == max ? argb & 0xFF : 0xFF );
            return qRgba( r, g, b, a );
        }

```
and then

```
        namespace integral
        {
            QSK_EXPORT constexpr QRgb operator""_rgba(
                const char* const str, const size_t len ) noexcept
            {
                return fromHexRgba( str, len );
            }
        }

        namespace color
        {
            QSK_EXPORT constexpr QColor operator""_rgba(
                const char* const str, const size_t len ) noexcept
            {
                const auto c = fromHexRgba( str, len );
                return { qRed( c ), qGreen( c ), qBlue( c ), qAlpha( c ) };
            }
    }
```
Guess it is possible to add a 3rd parameter - indicating the order - to fromHexString, so that we do not need fromHexRgba anymore:

```
           QSK_EXPORT constexpr QRgb operator""_rgba(
                const char* const str, const size_t len ) noexcept
            {
                return fromHexString( str, len, RGBA );
            }

           QSK_EXPORT constexpr QRgb operator""_argb(
                const char* const str, const size_t len ) noexcept
           {
                return fromHexString( str, len, ARGB );
           }
```